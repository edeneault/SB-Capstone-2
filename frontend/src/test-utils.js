import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
