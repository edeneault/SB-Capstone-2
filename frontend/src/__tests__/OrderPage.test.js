import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render, screen } from "../test-utils";
import OrderPage from "../pages/OrderPage.js";
import { createMemoryHistory, createLocation } from "history";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  const history = createMemoryHistory();
  const path = `/orders/:id`;

  const match = {
    isExact: false,
    path,
    url: path.replace(":id", "6125d0c2d3ffd29bedee24af"),
    params: { id: "6125d0c2d3ffd29bedee24af" },
  };

  test("renders OrderPage without crashing", () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <OrderPage match={match} history={history} />
      </Provider>,
      { initialRoutes: [`/orders/6125d0c2d3ffd29bedee24af`] },
    );
    // screen.debug();
    expect(getByText(/Order:/i)).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <OrderPage match={match} history={history} loading={false} />
      </Provider>,
      { initialRoutes: [`/orders/6125d0c2d3ffd29bedee24af`] },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
