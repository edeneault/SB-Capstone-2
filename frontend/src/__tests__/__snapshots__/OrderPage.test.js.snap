// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-5 pt-3 container-fluid"
  >
    <h1>
      Order: 6125d0c2d3ffd29bedee24af 
    </h1>
     
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        <div
          class="list-group list-group-flush"
        >
          <div
            class="list-group-item"
          >
            <h2>
              Shipping
            </h2>
            <p>
              <strong>
                Name: 
              </strong>
               John Doe
            </p>
            <p>
              <strong>
                Email: 
              </strong>
              <a
                href="mailto:g@example.com"
              >
                g@example.com
              </a>
            </p>
            <p>
              <strong>
                Address:
              </strong>
               9620 W. Russell Rd   APT 1025, Las Vegas, 89148, United States
            </p>
            <div
              class="fade rounded alert alert-danger show"
              role="alert"
            >
              Not Delivered
            </div>
          </div>
          <div
            class="list-group-item"
          >
            <h2>
              Payment Method
            </h2>
            <p>
              <strong>
                Method: 
              </strong>
            </p>
            <div
              class="fade rounded alert alert-success show"
              role="alert"
            >
               Paid on 
            </div>
          </div>
          <div
            class="list-group-item"
          >
            <h2>
              Order Items
            </h2>
            <div
              class="list-group list-group-flush"
            >
              <div
                class="list-group-item"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-md-1"
                  >
                    <img
                      alt="Lavazza - Espresso - Medium Roast"
                      class="img-fluid rounded"
                      src="https://res.cloudinary.com/edeneault-cloudinary/image/upload/v1630783263/eShop/sqiwu5ummm4y1to3cp5o.jpg"
                    />
                  </div>
                  <div
                    class="col"
                  >
                    <a
                      href="/products/6125d04db949259bdec4de3c"
                    >
                      Lavazza - Espresso - Medium Roast
                    </a>
                  </div>
                  <div
                    class="col"
                  >
                    1 x $9.99 = $9.99
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-4"
      >
        <div
          class="rounded card"
        >
          <div
            class="list-group list-group-flush"
          >
            <div
              class="list-group-item"
            >
              <h2>
                Order Summary
              </h2>
            </div>
            <div
              class="list-group-item"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Items
                </div>
                <div
                  class="col"
                >
                  $9.99
                </div>
              </div>
            </div>
            <div
              class="list-group-item"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Shipping
                </div>
                <div
                  class="col"
                >
                  $0
                </div>
              </div>
            </div>
            <div
              class="list-group-item"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Tax
                </div>
                <div
                  class="col"
                >
                  $1.5
                </div>
              </div>
            </div>
            <div
              class="list-group-item"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Total
                </div>
                <div
                  class="col"
                >
                  $11.49
                </div>
              </div>
            </div>
            <div
              class="list-group-item"
            >
              <a
                class=""
                href="/paymentmethod"
              >
                Change Payment Method
              </a>
            </div>
            <div
              class="list-group-item"
            >
              <div
                class="d-grid col"
              >
                <button
                  class="btn btn-small btn-dark btn btn-primary"
                  type="button"
                >
                  Mark as Delivered
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
